File: MyQuestions/AvenierEncoding February 2020/Prelims/ECFBP203.py
from collections import defaultdict, deque
from itertools import permutations
from sys import stdin,stdout
from bisect import bisect_left, bisect_right
from copy import deepcopy
from random import randint,randrange,choice

int_input=lambda : int(stdin.readline())
string_input=lambda : stdin.readline()
multi_int_input =lambda : map(int, stdin.readline().split())
multi_input = lambda : stdin.readline().split()
list_input=lambda : list(map(int,stdin.readline().split()))
string_list_input=lambda: list(string_input())

test = int_input()

for _ in range(test):
    n = int_input()
    arr = list_input()
    brr = [0 for i in range(n)]

    ptr = 0
    for i in range(n):
        if arr[i]!=0:
            brr[ptr] = arr[i];
            ptr+=1

    print(*brr)

File: MyQuestions/AvenierEncoding February 2020/Prelims/ECFBP203.java
import java.util.*;

class ECFBP203{

    static Scanner sc = new Scanner(System.in);
    public static void main(String args[]){
        int test = sc.nextInt();
        while(test-- >0){
            int n = sc.nextInt();
            int arr[] = new int[n];
            int brr[] = new int[n];
            Arrays.fill(brr,0);

            int ptr = 0;
            
            // imput array
            for(int  i = 0;i<n;i++){
                arr[i] = sc.nextInt();
            }

            // operate
            for(int i = 0;i<n;i++){
                if(arr[i]!=0){
                    brr[ptr++] = arr[i];
                }
            }

            for(int i = 0;i<n;i++){
                System.out.print(brr[i]+" ");
            }
            System.out.println();
        }
    }
}